py/mockbuild/buildroot.py

```diff
@@ -175,17 +175,17 @@ def doChroot(self, command, shell=False, nosync=False, *args, **kargs):
         env = dict(self.env)
         if nosync and self.nosync_path:
             env['LD_PRELOAD'] = self.nosync_path
-        if util.USE_NSPAWN:
+        if util.USE_NSPAWN or kargs.get('uid', 1000) == 0:
```

Comment:
I meant - why are you testing for uid == 1000? The uid of user can be different. E.g. 1005. I understand the change in backend.py. But this one I cannot comprehend.